server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadataBatch.java

```diff
@@ -329,7 +329,13 @@ public static class NodeStoreFilesMetadataBatch extends BaseNodeResponse {
 
         protected NodeStoreFilesMetadataBatch(StreamInput in) throws IOException {
             super(in);
-            this.nodeStoreFilesMetadataBatch = in.readMap(ShardId::new, NodeStoreFilesMetadata::new);
+            this.nodeStoreFilesMetadataBatch = in.readMap(ShardId::new, i -> {
+                if (i.readBoolean()) {
```

Comment:
these changes were merged in ShardBatchCache PR right? have you rebased from main?

Reply:
This commit was added by [here](https://github.com/opensearch-project/OpenSearch/pull/8746/commits/9b1d5536ee643162aa0c2cb450424e10236076e4) by @amkhar , because testNBatchesCreationAndAssignment() was failing.